File: pkg/signs/SignatureGenerator.cs
public class SignatureGenerator
    {
        public static string Generate(Dictionary<string, string> paramsDict, string apiKey)
        {
            // 对参数名进行排序
            var sortedKeys = paramsDict.Keys.OrderBy(k => k).ToList();

            // 拼接参数名和参数值，同时进行编码
            StringBuilder paramStr = new StringBuilder();
            foreach (var key in sortedKeys)
            {
                var value = paramsDict[key];
                if (!string.IsNullOrEmpty(value))
                {
                    var encodedKey = Uri.EscapeDataString(key);
                    var encodedValue = Uri.EscapeDataString(value);
                    if (paramStr.Length > 0)
                    {
                        paramStr.Append("&");
                    }
                    paramStr.Append(encodedKey).Append("=").Append(encodedValue);
                }
            }

            // 创建 HMAC - SHA256 实例
            using (HMACSHA256 hmac = new HMACSHA256(Encoding.UTF8.GetBytes(apiKey)))
            {
                // 计算签名
                byte[] hashBytes = hmac.ComputeHash(Encoding.UTF8.GetBytes(paramStr.ToString()));

                // 将字节数组转换为十六进制字符串
                StringBuilder hexString = new StringBuilder();
                foreach (byte b in hashBytes)
                {
                    hexString.Append(b.ToString("x2"));
                }

                return hexString.ToString();
            }
        }
    }

File: pkg/signs/SignatureGenerator.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.util.*;

public class SignatureGenerator {

    public static String generate(Map<String, String> params, String apiKey)
            throws NoSuchAlgorithmException, InvalidKeyException, UnsupportedEncodingException {
        // 对参数名进行排序
        final String paramStr = params.keySet()
                .stream()
                .sorted()
                .map(key -> {
                    try {
                        String encodedKey = URLEncoder.encode(key, StandardCharsets.UTF_8.name());
                        String encodedValue = URLEncoder.encode(params.getOrDefault(key, ""), StandardCharsets.UTF_8.name());
                        return encodedKey + "=" + encodedValue;
                    } catch (UnsupportedEncodingException e) {
                        // ignore
                    }
                    return "";
                })
                .filter(v -> !v.isEmpty())
                .collect(Collectors.joining("&"));
        // 创建 HMAC - SHA256 实例
        final Mac hmacSha256 = Mac.getInstance("HmacSHA256");
        hmacSha256.init(new SecretKeySpec(apiKey.getBytes(StandardCharsets.UTF_8), "HmacSHA256"));
        // 将字节数组转换为十六进制字符串
        final StringBuilder hexString = new StringBuilder();
        for (byte b : hmacSha256.doFinal(paramStr.getBytes(StandardCharsets.UTF_8))) {
            String hex = Integer.toHexString(0xff & b);
            if (hex.length() == 1) {
                hexString.append('0');
            }
            hexString.append(hex);
        }
        return hexString.toString();
    }
}
